grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -407,7 +414,11 @@ class UserService {
 
     public def getEffectiveRoles(User user) {
         def currentLocation = AuthService.currentLocation?.get()
-        return user.getEffectiveRoles(currentLocation)
+        return getEffectiveRoles(user, currentLocation)
+    }
+
+    public def getEffectiveRoles(User user, Location location) {
+        return user.getEffectiveRoles(location)
```

Comment:
I don't mind that, but was it necessary? I think the method above was clean enough?

Reply:
If you are wondering why I split the `getEffectiveRoles()` methods, it is because previously it defaulted to current location and I needed roles for a user in origin location so I added an additional method which  receives a location parameter.

Reply:
ok, I did not notice that there was a currentLocation :+1: